### ISSUE TITLE
[Question] Filter elements where an attribute does not exist 

### ISSUE THREAD

**Issue by mkp-gebensleben** [FraunhoferIOSB/FROST-Server#573]
Is it possible to find objects where a certain field/attribute does not exist?

For example, I want to get all Observations that don't have _parameters/demo = <any value>.

Thank you!

**Comment by hylkevds** [comment]
Unfortunately, at the moment you can't do this directly, since null values never match. This is why the SQL language has the specific `is null` construct, but that is not implemented in STA / FROST.

OData does have the `null` keyword for comparisons. I'll probably implement that at some point, since it's very useful.

A sort-of workaround is to $orderby on the field in descending order. Since null values come last in ascending order, they come first in descending order. Of course this also gives you all the non-null values, but at least you can see the null values grouped together.

**Comment by mkp-gebensleben** [comment]
Thank you!

Is there a difference between null values and values that are actually not there with respect to filtering?

In my case, parameters/demo is not explicitly null when I'm inspecting the Observations, it's completely absent.

**Comment by hylkevds** [comment]
No, it does not differentiate between json `null` and "not there".

**Comment by mkp-gebensleben** [comment]
Thank you for the info.

Unfortunately in our case ordering won't help much. We will have to tag every Observation with `parameters/demo = true` or `false`.

**Comment by mjacoby** [comment]
+1 I also just came across an application scenario where filtering by is null/is not null would be quite helpful.
